api/include/opentelemetry/trace/span.h

```diff
@@ -36,11 +37,13 @@ struct StartSpanOptions
   core::SystemTimestamp start_system_time;
   core::SteadyTimestamp start_steady_time;
 
+  // Optionally set attributes at Span creation.
+  nostd::span<std::pair<nostd::string_view, common::AttributeValue>> attributes;
```

Comment:
Do we de-dupe the names? If yes, might need to clarify, it could be one of the following:
1. the last one will win (overwrite the previous ones).
2. the span creation would fail (end up having invalid span).
3. the first one will win (subsequent attributes with the same name will be ignored).

Reply:
I went with (1), which in my reading is most in line with the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#set-attributes):
> Setting an attribute with the same key as an existing attribute SHOULD overwrite the existing attribute's value.

Reply:
Another note on this: it is my understanding that `std::pair` is ABI stable, but please anybody correct me if I'm wrong with that.

Reply:
I don't know if pair is stable across different STLs, but adding a struct for { key, value } might even increase readability.

Reply:
I added a `AttributeKeyValue`, which indeed makes this much better.

Reply:
After re-evaluating this over the weekend, I switched to using a `KeyValueIterable` as function argument, as we do for `AddEvent`. This should take care of any lifetime issues.

On the flipside, we might end up with some more `StartSpan` signatures when we finally support links.